Clarify how to use the Galaxy token (#65961)
ansible/ansible: docs/docsite/rst/dev_guide/developing_collections.rst

Getting your token or API key
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

To upload your collection to Galaxy, you must first obtain an API token (``--api-key`` in the ``ansible-galaxy`` CLI command or ``token`` in the :file:`ansible.cfg` file under the ``galaxy_server`` section). The API token is a secret token used to protect your content.

To get your API token:

* For Galaxy, go to the `Galaxy profile preferences <https://galaxy.ansible.com/me/preferences>`_ page and click :guilabel:`API token`.
* For Automation Hub, go to https://cloud.redhat.com/ansible/automation-hub/token/ and click :guilabel:`Get API token` from the version dropdown.

Storing or using your API token
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Once you have retrieved your API token, you have the following options on how you can store or use that token for collections:

Reviewer: ```suggestion
Once you have retrieved your API token, you can store or use the token for collections in two ways:
```

Getting your token or API key
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

To upload your collection to Galaxy, you must first obtain an API token (``--api-key`` in the ``ansible-galaxy`` CLI command or ``token`` in the :file:`ansible.cfg` file under the ``galaxy`` section). The API token is a secret token used to protect your content.

To get your API token:

* For galaxy, go to the `Galaxy profile preferences <https://galaxy.ansible.com/me/preferences>`_ page and click :guilabel:`API token`.
* For Automation Hub, go to https://cloud.redhat.com/ansible/automation-hub/token/ and click :guilabel:`Get API token` from the version dropdown.

Storing or using your API token
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Once you have retrieved your AP token, you have three options on how you can store or use that token for collections:

* Pass the token to  the ``ansible-galaxy`` command using the ``--api-key``.

> * Save the token in a Galaxy token file.

Reviewer 2: I'm a bit hesitant to document this, if `ansible-galaxy login` ever goes away then this would probably go with it. IMO we should only have the `--api-key` and entries in `ansible.cfg`.

Author: removed.

.. code-block:: bash

    ansible-galaxy collection publish ./geerlingguy-collection-1.2.3.tar.gz --api-key=[key goes here]

Use a galaxy token file
.......................

You can store your API token in your a Galaxy token file (in conjunction with the ``--server`` argument or :ref:`GALAXY_SERVER` setting in your :file:`ansible.cfg` file). This file is :file:`~/.ansible/galaxy_token` by default. Use the :ref:`GALAXY_TOKEN_PATH` to tell the ``ansible-galaxy`` command to find the token file at some other location. You cannot use the token file with any servers defined in your :ref:`Galaxy server list <galaxy_server_config>`.

1. Save the token in the default token file (:file:`~/.ansible/galaxy_token`).
2. Publish to the default server:

  .. code-block:: bash

      ansible-galaxy collection publish ./geerlingguy-collection-1.2.3.tar.gz

3. Alternately, publish to another server:

  .. code-block:: bash

>       ansible-galaxy collection publish --server my_private_galaxy ./geerlingguy-collection-1.2.3.tar.gz

Reviewer 2: This particular scenario would have `--server` be a URL.

Author: removed

Specify the token within a Galaxy server list
.............................................

With this option, you configure one or more servers for Galaxy in your :file:`ansible.cfg` file under the ``galaxy_server_list`` section. For each server, you also configure the token.

.. code-block:: ini

   [galaxy]
   server_list = release_galaxy

   [galaxy_server.release_galaxy]

>      url=https://galaxy.ansible.com/

Reviewer 2: Should this be indented more than `[galaxy_server.release_galaxy]`?

Author: doh!  thanks you are correct. Fixed.

.. code-block:: bash

     ansible-galaxy collection publish path/to/my_namespace-my_collection-1.0.0.tar.gz

.. note::

> 	The above command assumes you have retrieved and stored your API token in a file at the default token path or as part of a Galaxy server list. See :ref:`galaxy_get_token` for details.

Author: ```suggestion
	The above command assumes you have retrieved and stored your API token as part of a Galaxy server list. See :ref:`galaxy_get_token` for details.
```
